### PR TITLE
feat: implement custom mockes and drop go-metrics

### PR DESCRIPTION
**Pull request by kruskall**
## What does this PR do?

implement a custom mockes and get rid of go-metrics and related deps

## Why is it important?

go-metrics hasn't been updated in 4 years

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

Blocked by https://github.com/elastic/mock-es/pull/9

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: testing/integration/logs_ingestion_test.go [hunk 1]
```diff
@@ -110,16 +109,56 @@ func TestLogIngestionFleetManaged(t *testing.T) {
...
-		uid,
-		clusterUUID,
-		registry,
-		time.Now().Add(time.Hour), 0, 0, 0, 100, 0))
+	mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("X-Elastic-Product", "Elasticsearch")
+		w.Header().Set(http.CanonicalHeaderKey("Content-Type"), "application/json")
+		switch {
+		case r.Method == http.MethodGet && r.URL.Path == "/":
+			root := fmt.Sprintf("{\"name\" : \"mock\", \"cluster_uuid\" : \"%s\", \"version\" : { \"number\" : \"0.0.0\", \"build_flavor\" : \"default\"}}", clusterUUID)
```
Comment (swiatekm) [inline testing/integration/logs_ingestion_test.go:121]: These raw JSON test templates are very ugly. Couldn't we use a map and just serialize it to JSON?

File: testing/integration/logs_ingestion_test.go [hunk 2]
```diff
@@ -110,16 +109,56 @@ func TestLogIngestionFleetManaged(t *testing.T) {
...
+				rd, err := gzip.NewReader(body)
+				if err != nil {
+					panic(err)
+				}
+				defer rd.Close()
+				body = rd
+			}
+
+			var b []byte
+			b = append(b, []byte(`{"items":[`)...)
```
Comment (swiatekm) [inline testing/integration/logs_ingestion_test.go:136]: Same as above, this looks very fragile to me. I'd prefer to use slices and maps here, and serialize to JSON at the end.

**Review by leehinman — COMMENTED** [review]
If the fact that `go-metrics` hasn't been updated in 4 years is really the "problem", can you please just propose a fix in the mock-es repo?

mocking elasticsearch is complicated enough (magic headers, license format, UUID requirements, etc) that I really don't want to see this duplicated all over the place in our testing code.

**Comment by blakerouse** [comment]
I agree with @leehinman on this one. This is drastically increasing the test logic, when it was much simpler.

Another question is it not being updated an issue? Does it have a CVE or something in it that requires it to be updated?

**Comment by leehinman** [comment]
@kruskall I finally got your changes merged to mock-es, do you want to close this PR or re-work it to pull in new mock-es?

**Review by leehinman — APPROVED** [review]

**Pull request merged**